### ISSUE TITLE
expanded_extent_ == nullptr || isBroadcast() INTERNAL ASSERT FAILED 

### ISSUE THREAD

**Issue by yushangdi** [pytorch/pytorch#82858]
### 🐛 Describe the bug

I'm running the code below with flag `PYTORCH_NVFUSER_DISABLE=fallback`, and got error below. I'm not sure what's the problem? Thanks!

Traceback (most recent call last):
  File "microbench.py", line 77, in <module>
    script_f(*arg_list)
  File "/scratch/shangdiy/work/pytorch/torch/nn/modules/module.py", line 1186, in _call_impl
    return forward_call(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: expanded_extent_ == nullptr || isBroadcast() INTERNAL ASSERT FAILED at "/scratch/shangdiy/work/pytorch/torch/csrc/jit/codegen/cuda/ir_internal_nodes.h":858, please report a bug to PyTorch. Expanded extent is only relevant for strided broadcast dimensions yet found an expanded extent without a strided broadcast iter type.

```python
from torchvision.models import resnet18
import torch.utils._pytree as pytree
from torch.nn.utils.stateless import functional_call
from functorch.compile import ts_compile
from functorch.compile import default_decompositions
import torch.utils._pytree as pytree
from torchdynamo.testing import reduce_to_scalar_loss


def trace_model(model, inputs):
    """
    Get the full graph (both forward and backward) of `model` on `inputs`
    The moddel should have a single forward and a single backward graph
    """
    def f(params, inp):
        out = functional_call(model, params, inp)
        loss = reduce_to_scalar_loss(out)
        loss.backward()
        return [param.grad for param in params.values()]
    
    params = dict(model.named_parameters())
    traced_graph = make_fx(f, decomposition_table=default_decompositions)(params, inputs)
    return traced_graph, params

inputs = torch.rand(1, 3, 224, 224, device="cuda")
model = resnet18(pretrained=True)
model.to("cuda")


model.eval()
traced_graph, params = trace_model(model, inputs)
traced_graph.graph.set_codegen(torch.fx.graph.CodeGen())  # avoid recursive pytree
arg_list, spec  = pytree.tree_flatten([params, inputs])
# print(traced_graph)
script_f = ts_compile(traced_graph, 0)
with torch.no_grad():
    script_f(*arg_list)
    script_f(*arg_list)
```

### Versions

Collecting environment information...
PyTorch version: 1.13.0a0+git50c655d
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (conda-forge gcc 12.1.0-16) 12.1.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.5
Libc version: glibc-2.27

Python version: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-1069-aws-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 11.6.112
GPU models and configuration: 
GPU 0: NVIDIA A100-SXM4-40GB
GPU 1: NVIDIA A100-SXM4-40GB
GPU 2: NVIDIA A100-SXM4-40GB
GPU 3: NVIDIA A100-SXM4-40GB
GPU 4: NVIDIA A100-SXM4-40GB
GPU 5: NVIDIA A100-SXM4-40GB
GPU 6: NVIDIA A100-SXM4-40GB
GPU 7: NVIDIA A100-SXM4-40GB

Nvidia driver version: 510.47.03
cuDNN version: Probably one of the following:
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] botorch==0.6.4
[pip3] flowtorch==0.8
[pip3] functorch==0.3.0a0+9887d51
[pip3] gpytorch==1.6.0
[pip3] mypy==0.960
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.0
[pip3] numpyro==0.9.2
[pip3] pytorch-transformers==1.2.0
[pip3] torch==1.13.0a0+git9887d51
[pip3] torch-struct==0.5
[pip3] torchaudio==0.12.0a0+4d2fa19
[pip3] torchdynamo==1.13.0.dev0
[pip3] torchfile==0.1.0
[pip3] torchmetrics==0.9.1
[pip3] torchrec-nightly==2022.4.26
[pip3] torchtext==0.13.0a0+d6e3550
[pip3] torchvision==0.14.0a0+a7e4fbd
[pip3] torchx-nightly==2022.6.8
[conda] blas                      1.0                         mkl  
[conda] botorch                   0.6.4                    pypi_0    pypi
[conda] flowtorch                 0.8                      pypi_0    pypi
[conda] functorch                 0.3.0a0+9887d51           dev_0    <develop>
[conda] gpytorch                  1.6.0                    pypi_0    pypi
[conda] magma-cuda110             2.5.2                         1    pytorch
[conda] magma-cuda116             2.6.1                         0    pytorch
[conda] mkl                       2022.1.0                 pypi_0    pypi
[conda] mkl-include               2022.0.1           h06a4308_117  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.1            py38h8666266_1    conda-forge
[conda] mkl_random                1.2.2            py38h1abd341_0    conda-forge
[conda] numpy                     1.21.2                   pypi_0    pypi
[conda] numpyro                   0.9.2                    pypi_0    pypi
[conda] pytorch-transformers      1.2.0                    pypi_0    pypi
[conda] torch                     1.13.0a0+git9887d51           dev_0    <develop>
[conda] torch-struct              0.5                      pypi_0    pypi
[conda] torchaudio                0.12.0a0+4d2fa19           dev_0    <develop>
[conda] torchdynamo               1.13.0.dev0               dev_0    <develop>
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchmetrics              0.9.0                    pypi_0    pypi
[conda] torchrec-nightly          2022.4.26                pypi_0    pypi
[conda] torchtext                 0.13.0a0+d6e3550           dev_0    <develop>
[conda] torchvision               0.14.0a0+a7e4fbd           dev_0    <develop>
[conda] torchx-nightly            2022.6.8                 pypi_0    pypi

**Comment by jjsjann123** [comment]
Looks like my patch for expand didn't work. :( We DO have another expand PR that's supposed to patch other issues. Let me try this repro there to double check.
Thanks a lot for reporting and tagging me on this~~

**Comment by yushangdi** [comment]
@jjsjann123 Thanks for looking into this! I also got the following error when running 
`PYTORCH_NVFUSER_DISABLE_FALLBACK=1 python benchmarks/torchbench.py --training --devices=cuda --nvfuser --accuracy-aot-ts-mincut --use-eval-mode --isolate --repeat 50 -k hf_Albert --float16` in torchdynamo. Is this a similar issue?


Traceback (most recent call last):
  File "benchmarks/torchbench.py", line 418, in <module>
    main(TorchBenchmarkRunner(), original_dir)
  File "/scratch/shangdiy/work/torchdynamo/benchmarks/common.py", line 1441, in main
    runner.run_one_model(
  File "/scratch/shangdiy/work/torchdynamo/benchmarks/common.py", line 870, in run_one_model
    model_iter_fn(model, example_inputs)
  File "benchmarks/torchbench.py", line 395, in forward_and_backward_pass
    def forward_and_backward_pass(self, mod, inputs, collect_outputs=True, stats=None):
  File "benchmarks/torchbench.py", line 395, in forward_and_backward_pass
    def forward_and_backward_pass(self, mod, inputs, collect_outputs=True, stats=None):
  File "benchmarks/torchbench.py", line 401, in forward_and_backward_pass
    pred = mod(*cloned_inputs)
  File "/scratch/shangdiy/work/pytorch/torch/nn/modules/module.py", line 1186, in _call_impl
    return forward_call(*input, **kwargs)
  File "/scratch/shangdiy/work/py38/lib/python3.8/site-packages/transformers/models/albert/modeling_albert.py", line 917, in forward
    @add_start_docstrings_to_model_forward(ALBERT_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
  File "/scratch/shangdiy/work/torchdynamo/torchdynamo/eval_frame.py", line 94, in _fn
    return fn(*args, **kwargs)
  File "/scratch/shangdiy/work/pytorch/functorch/functorch/_src/aot_autograd.py", line 721, in forward
    return compiled_f(
  File "/scratch/shangdiy/work/torchdynamo/torchdynamo/eval_frame.py", line 94, in _fn
    return fn(*args, **kwargs)
  File "/scratch/shangdiy/work/pytorch/functorch/functorch/_src/aot_autograd.py", line 278, in forward
    fw_outs = normalize_as_list(compiled_fw(*flat_tensor_args))
  File "/scratch/shangdiy/work/pytorch/torch/nn/modules/module.py", line 1186, in _call_impl
    return forward_call(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: !fallback_mode_enabled_ INTERNAL ASSERT FAILED at "/scratch/shangdiy/work/pytorch/torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp":925, please report a bug to PyTorch. Couldn't succcessfully sort out the fusion expressions. There are remaining connections of the heirarchical segmentation which should have been flattened to a single ordered group, or disjoint ordered groups.

**Comment by jjsjann123** [comment]
Verified that the original problem went away with this PR: https://github.com/csarofeen/pytorch/pull/1883
Let me try to cherry-pick that after we get it merged. Note that this would also have to wait until #83067 merged first...

**Comment by jjsjann123** [comment]
^^^ FYI, the torchdynamo issue does repro on our nightlies, but has been fixed in the lastest devel branch for nvfuser last week and I just verified that we have it patched in our latest upstream push #83067 
cc'ing @yushangdi

**Comment by yushangdi** [comment]
> ^^^ FYI, the torchdynamo issue does repro on our nightlies, but has been fixed in the lastest devel branch for nvfuser last week and I just verified that we have it patched in our latest upstream push #83067 cc'ing @yushangdi

Thanks!

**Comment by jjsjann123** [comment]
Looks like we have the issue closed already. Just a note that the original expand issue is going to be resolve in latest push with expand fixes: #83239
